test/e2e/csv_e2e_test.go

```diff
@@ -4204,29 +4204,78 @@ var _ = Describe("Disabling copied CSVs", func() {
 			},
 		},
 	}
+	BeforeEach(func() {
+		Eventually(func() error {
 
-	When("an operator is installed in AllNamespace mode", func() {
-		BeforeEach(func() {
-			Eventually(func() error {
-				if err := ctx.Ctx().Client().Create(context.TODO(), ns); err != nil && !k8serrors.IsAlreadyExists(err) {
-					ctx.Ctx().Logf("Unable to create ns: %v", err)
-					return err
-				}
+			ns = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: genName("csv-toggle-test-"),
+				},
+			}
 
-				if err := ctx.Ctx().Client().Create(context.TODO(), &operatorGroup); err != nil && !k8serrors.IsAlreadyExists(err) {
-					ctx.Ctx().Logf("Unable to create og: %v", err)
-					return err
-				}
+			operatorGroup = operatorsv1.OperatorGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      genName("csv-toggle-test-"),
+					Namespace: ns.GetName(),
+				},
+			}
 
-				if err := ctx.Ctx().Client().Create(context.TODO(), &csv); err != nil && !k8serrors.IsAlreadyExists(err) {
-					ctx.Ctx().Logf("Unable to create csv: %v", err)
-					return err
-				}
+			csv = operatorsv1alpha1.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      genName("csv-toggle-test-"),
+					Namespace: ns.GetName(),
+				},
+				Spec: operatorsv1alpha1.ClusterServiceVersionSpec{
+					InstallStrategy: newNginxInstallStrategy(genName("csv-toggle-test-"), nil, nil),
+					InstallModes: []operatorsv1alpha1.InstallMode{
+						{
+							Type:      operatorsv1alpha1.InstallModeTypeAllNamespaces,
+							Supported: true,
+						},
+					},
+				},
+			}
 
-				return nil
-			}).Should(Succeed())
-		})
+			if err := ctx.Ctx().Client().Create(context.TODO(), ns); err != nil && !k8serrors.IsAlreadyExists(err) {
+				ctx.Ctx().Logf("Unable to create ns: %v", err)
+				return err
+			}
+			if err := ctx.Ctx().Client().Create(context.TODO(), &operatorGroup); err != nil && !k8serrors.IsAlreadyExists(err) {
+				ctx.Ctx().Logf("Unable to create og: %v", err)
+				return err
+			}
 
+			if err := ctx.Ctx().Client().Create(context.TODO(), &csv); err != nil && !k8serrors.IsAlreadyExists(err) {
+				ctx.Ctx().Logf("Unable to create csv: %v", err)
+				return err
+			}
+
+			return nil
+		}).Should(Succeed())
+
+	})
+	AfterEach(func() {
+		Eventually(func() error {
+			err := ctx.Ctx().Client().Delete(context.TODO(), &csv)
+			if err != nil {
```

Comment:
question: are you sure you don't need to check for "NotFound" failures here as well?

Reply:
You are right.  I'll add it.  Thanks!